utam-compiler/src/main/java/utam/compiler/representation/ElementMethod.java

```diff
@@ -296,6 +300,15 @@ public Filtered(
       this.unionType = asUnionTypeOrNull(implType);
     }
 
+    private void setFilterParameterClassImports() {
+      // For non-primitive method parameters added by a filter, we must add class imports
+      // for those types or the generated code will not be compilable by javac.
+      parametersTracker.getMethodParameters().stream()
+          .filter(param -> Arrays.stream(PrimitiveType.values())
+              .noneMatch(primitiveType -> primitiveType.isSameType(param.getType())))
+          .forEach(param -> ParameterUtils.setImport(classImports, param.getType()));
```

Comment:
filter is redundant, ParameterUtils.setImport is not going to add imports for primitives

Reply:
I didn't notice that. Clever. That simplifies things quite a bit. Fixed in next commit.

Reply:
sorry, forgot but we should filter for parameter not being literal:
```
parametersTracker.getMethodParameters().
          stream().filter(p -> !p.isLiteral())
          .forEach(p -> setImport(classImports, p.getType()));
```

Reply:
No, I'm afraid that doesn't work. Consider:
```json
{
  "elements": [
    {
      "name": "test",
      "public": true,
      "selector": {
        "css": "my-custom-element[with-attribute='attr-value']",
        "returnAll": true
      },
      "filter": {
        "apply": "containsElement",
        "args": [
          {
            "type": "locator",
            "value": {
              "css": "input[value='%s']",
              "args": [
                {
                  "name": "value",
                  "type": "string"
                }
              ]
            }
          }
        ],
        "findFirst": true,
        "matcher": {
          "type": "isTrue"
        }
      }
    }
  ]
}
```
The `locator` parameter of the `containsElement` action for the filter _is_ a literal. If we filter on not-literal, we won't add the `LocatorBy` import, which is one of the whole reasons for this PR in the first place.